Sources/PowerSync/Kotlin/db/PowerSyncDataTypeConvertible.swift

```diff
@@ -0,0 +1,32 @@
+import struct Foundation.Data
+
+/// Represents the set of types that are supported
+/// by the PowerSync Kotlin Multiplatform SDK
+public enum PowerSyncDataType {
+    case bool(Bool)
+    case string(String)
+    case int64(Int64)
+    case int32(Int32)
+    case double(Double)
+    case data(Data)
+}
+
+/// Types conforming to this protocol will be
+/// mapped to the specified ``PowerSyncDataType``
+/// before use by the PowerSync Kotlin Multiplatform SDK
```

Comment:
Minor wording nit: We don't necessarily need to mention the Kotlin Multiplatform SDK here. This is generally used for SQLite query parameters.
```suggestion
/// before use by SQLite
```